src/ts/component/menu/dataview/template/list.tsx

```diff
@@ -100,8 +100,20 @@ class MenuTemplateList extends React.Component<I.Menu> {
 	};
 
 	componentDidMount () {
+		const { param } = this.props;
+		const { data } = param;
+		const { rootId, blockId, typeId, templateId, getView, hasSources } = data;
+		const view = getView();
+
 		this.rebind();
-		this.load(true);
+
+		UtilObject.checkDefaultTemplate(typeId, templateId, (res) => {
+			if (!hasSources || !res) {
+				C.BlockDataviewViewUpdate(rootId, blockId, view.id, { ...view, defaultTemplateId: '' }, () => this.load(true));
```

Comment:
what is the purpose of updating single view if the template is incorrect, there is no sense in doing so, you should not show it as selected in the rendering phase without updating views